tools/total-contributions

```diff
@@ -0,0 +1,170 @@
+directory, which does not need to be the directory this is run from.
+
+# Changes between two major releases.
+total-contributions 4.0 5.0
+
+# Changes between a release and the current main branch.
+total-contributions 4.0 main
+total-contributions 2.1.0
+    """,
+)
+parser.add_argument(
+    "version",
+    metavar="version",
+    nargs="*",
+    # TODO: Ideally, we'd replace "1.3.0" with "First commit", to
+    # simplify including contributions before the 1.3.0 release.
+    default=["1.3.0", "main"],
+    help="Git tag or branch in zulip/zulip specifying one end of the commit range to use.",
+)
+
+parser.add_argument(
+    "-a",
+    "--ascending",
+    action="store_true",
+    help="Sort contributors based on number of commits(ascending order)",
+)
+
+args = parser.parse_args()
+
+if len(args.version) > 2:
+    parser.error("Expects 0 to 2 version number(s)")
+
+lower_zulip_version = args.version[0]
+if len(args.version) == 1:
+    upper_zulip_version = "main"
+else:
+    upper_zulip_version = args.version[1]
+
+subprocess.check_call(["git", "fetch"], cwd=find_path("zulip"))
+
+# Extract git version and time. It's important that we use the commit
+# date (%ci), not the author date (%ai), since while those are often
+# near identical for release commits, if we pass a branch like `main`,
+# it's possible the latest commit on the branch might have a months
+# old author date if the last pull request merged was started at that
+# time.
+try:
+    lower_time = subprocess.check_output(
+        ["git", "log", "-1", "--format=%ci", lower_zulip_version],
+        stderr=subprocess.DEVNULL,
+        text=True,
+    ).split()[0]
+    upper_time = subprocess.check_output(
+        ["git", "log", "-1", "--format=%ci", upper_zulip_version],
+        stderr=subprocess.DEVNULL,
+        text=True,
+    ).split()[0]
+except subprocess.CalledProcessError:
+    print("Specified version(s) don't exist")
+    sys.exit(0)
+
+out_dict: Dict[str, int] = {}
+zulip = retrieve_log("zulip", lower_zulip_version, upper_zulip_version)
+add_log(out_dict, zulip)
+
+# TODO: This should probably include more repositories in the zulip organization.
+for repository in [
+    "zulip-mobile",
+    "zulip-desktop",
+    "docker-zulip",
+    "python-zulip-api",
+    "zulip-terminal",
+]:
+    subprocess.check_call(["git", "fetch"], cwd=find_path(repository))
+    lower_repo_version = find_version(lower_time, repository)
+    upper_repo_version = find_version(upper_time, repository)
+    repo_log = retrieve_log(repository, lower_repo_version, upper_repo_version)
+    add_log(out_dict, repo_log)
+
+# Sorting based on number of commits
+if args.ascending:
+    for commit_count, committer_name in sorted(out_dict.items(), key=lambda keys: keys[1]):
+        print(str(committer_name) + "\t" + commit_count)
+else:
+    for commit_count, committer_name in sorted(
+        out_dict.items(), key=lambda keys: keys[1], reverse=True
```

Comment:
nit: `keys` is an inaccurate name for the parameters to these two lambdas; each argument is one key and one value, not a collection of keys.

One good name would be `item`.

Reply:
Changed.